python/cudf/cudf/tests/test_rank.py

```diff
@@ -10,152 +10,145 @@
-            gdf["col2"],
-            pdf["col2"],
-            method=method,
-            na_option=na_option,
-            ascending=ascending,
-            pct=pct,
-        )
-        # TODO: https://github.com/pandas-dev/pandas/issues/32593
-        # Dataframe (bug in pandas)
-        # _check(
-        #     gdf,
-        #     pdf,
-        #     method=method,
-        #     na_option=na_option,
-        #     ascending=ascending,
-        #     pct=pct,
-        # )
-
-    def test_rank_error_arguments(self):
-        pdf = pd.DataFrame(index=self.index)
-        pdf["col1"] = self.col1
-        pdf["col2"] = self.col2
-        gdf = DataFrame.from_pandas(pdf)
-
-        assert_exceptions_equal(
-            lfunc=pdf["col1"].rank,
-            rfunc=gdf["col1"].rank,
-            lfunc_args_and_kwargs=(
-                [],
-                {
-                    "method": "randomname",
-                    "na_option": "keep",
-                    "ascending": True,
-                    "pct": True,
-                },
+@pytest.fixture
+def pdf():
+    return pd.DataFrame(
+        {
+            "col1": np.array([5, 4, 3, 5, 8, 5, 2, 1, 6, 6]),
+            "col2": np.array(
+                [5, 4, np.nan, 5, 8, 5, np.inf, np.nan, 6, -np.inf]
             ),
-            rfunc_args_and_kwargs=(
-                [],
-                {
-                    "method": "randomname",
-                    "na_option": "keep",
-                    "ascending": True,
-                    "pct": True,
-                },
-            ),
-        )
+        },
+        index=np.array([5, 4, 3, 2, 1, 6, 7, 8, 9, 10]),
+    )
 
-        assert_exceptions_equal(
-            lfunc=pdf["col1"].rank,
-            rfunc=gdf["col1"].rank,
-            lfunc_args_and_kwargs=(
-                [],
-                {
-                    "method": "first",
-                    "na_option": "randomname",
-                    "ascending": True,
-                    "pct": True,
-                },
-            ),
-            rfunc_args_and_kwargs=(
-                [],
-                {
-                    "method": "first",
-                    "na_option": "randomname",
-                    "ascending": True,
-                    "pct": True,
-                },
-            ),
+
+@pytest.mark.parametrize("dtype", ["O", "f8", "i4"])
+@pytest.mark.parametrize("ascending", [True, False])
+@pytest.mark.parametrize("method", ["average", "min", "max", "first", "dense"])
+@pytest.mark.parametrize("na_option", ["keep", "top", "bottom"])
+@pytest.mark.parametrize("pct", [True, False])
+@pytest.mark.parametrize("numeric_only", [True, False])
+def test_rank_all_arguments(
+    pdf, dtype, ascending, method, na_option, pct, numeric_only
+):
+    if method == "first" and dtype == "O":
+        # not supported by pandas
+        return
+    if numeric_only:
+        pdf["str"] = np.array(
+            ["a", "b", "c", "d", "e", "1", "2", "3", "4", "5"]
```

Comment:
does it break everything if we just add this to the `pdf` fixture ?

Reply:
Yeah good catch - the test cases don't fail probably because all threads are adding the same column to `pdf`, but that's an explicit race condition. I'll make a local deep copy to `pdf` before this step.